src/main.cc

```diff
@@ -4,14 +4,53 @@
 
 #include <string.h>
 
+#include <windows.h>
+#include <winver.h>
+
 #include "rescle.h"
-#include "version.h"
 
 namespace {
 
-void print_help() {
+LPVOID get_file_version_info() {
+  DWORD zero = 0;
+  DWORD filename_buffer_size = MAX_PATH;
+  LPWSTR filename = nullptr;
+  SetLastError(ERROR_SUCCESS);
+
+  do {
+    filename = new wchar_t[filename_buffer_size];
+
+    GetModuleFileNameW(NULL, filename, filename_buffer_size);
+
+    // Double the buffer size in case the path is longer
+    filename_buffer_size *= 2;
+  } while (GetLastError() == ERROR_INSUFFICIENT_BUFFER);
+
+  if (GetLastError() != ERROR_SUCCESS) {
+    return nullptr;
+  }
```

Comment:
```suggestion
  std::vector<wchar_t> filename(MAX_PATH);
  do {
    GetModuleFileNameW(NULL, filename.data(), filename.size());

    // Double the buffer size in case the path is longer
    filename.resize(filename.size() * 2);
  } while (GetLastError() == ERROR_INSUFFICIENT_BUFFER);

  if (GetLastError() != ERROR_SUCCESS) {
    return nullptr;
  }
```